### ISSUE TITLE
[hexagon] Add hexagon tests to lint 

### ISSUE THREAD

**Issue by mehrdadh** [apache/tvm#12579]
Let's add hexagon tests to link step to avoid issues like this:
https://github.com/apache/tvm/pull/12565

**Comment by mehrdadh** [comment]
@quic-sanirudh I see that you claimed almost all hexagon tests, please let me know if you want to offload some.

https://github.com/apache/tvm/issues/11414

**Comment by quic-sanirudh** [comment]
> @quic-sanirudh I see that you claimed almost all hexagon tests, please let me know if you want to offload some.
> 
> #11414

@mehrdadh Yes I took it up to do it in my spare time and I've pushed a couple PRs to fix linter for a few files, but haven't started working on the rest. Please feel free to pick up any of the remaining ones, thanks.

**Comment by masahi** [comment]
I think this has been completed.
